bazel/grpc_build_system.bzl

```diff
@@ -35,6 +35,18 @@ load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
 # The set of pollers to test against if a test exercises polling
 POLLERS = ["epollex", "epoll1", "poll"]
 
+# The set of known EventEngines to test
+EVENT_ENGINES = [
+    {
+        "name": "libuv",
+        "tags": [],
+    },
+    {
+        "name": "poll",
+        "tags": ["no_windows", "no_mac"],
```

Comment:
Why not test poll on mac?

Reply:
My reasoning was A) it's unlikely this engine will be used beyond a few known scenarios which are non-mac, and B) the expense of testing an extra ~350 end2end tests (so far), measured mostly in time waiting for CI to finish. It may also be easier to implement if there are no guarantees that it must also work on mac. That's not a hard requirement at the moment.

That said, we could test it, I'm just not entirely sure it's worth the cost(s).

Reply:
I think it would be good to enumerate the costs before prematurely optimizing. In my experience testing on multiple platforms is useful in shaking out bad assumptions.

Reply:
The costs are extremely high here.  The optimization is absolutely worthwhile.

Reply:
Oh, wait, sorry -- I thought this comment was about why we don't do the cross-product of pollers and EventEngines.

Sounds like you're just asking about specifically why we don't run the poll-based EventEngine on MacOS.  I agree that that would be useful.

Reply:
The costs are enumerated, but not measured. It is still a multiplier, adding another test on mac for every test where `uses_polling=True`. The cost is roughly equivalent to adding another iomgr poller. We expect a decent CI speed/agility gain by eliminating the (eventually) redundant iomgr poller tests, and testing another EventEngine on mac is equivalent to *not* removing one of the iomgr pollers.

I agree it would be valuable to test it on mac, no argument. But I'm not sure it's worth the cost(s).

Reply:
If we are actually going to support this as a general-purpose OSS EventEngine, I do think that we will want to test it on MacOS.

Reply:
Ok. Given there's a short timeline for an implementation that does not require mac to work, would it be a fair compromise to leave a `TODO(tamird)` to enable mac tests when ready, and not block submission if there are mac-specific quirks to work out?

Reply:
This configuration will need to be done internally, since the affected implementations now live elsewhere.